puzzle/blueprints/public/views.py

```diff
@@ -122,3 +126,45 @@ def delete_genelist(list_id, case_id=None):
         # remove the whole gene list
         app.db.remove_genelist(list_id)
         return redirect(url_for('.index'))
+
+
+@blueprint.route('/resources', methods=['POST'])
+def resources():
+    """Upload a new resource for an individual."""
+    ind_id = request.form['ind_id']
+
+    upload_dir = os.path.abspath(app.config['UPLOAD_DIR'])
+    req_file = request.files['file']
```

Comment:
Flask will actually catch the key errors and return the correct status code but I guess it would be nice to return the user to the form with some more customized error message - will look into it later :)